UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/BaseController.cs

```diff
@@ -33,6 +36,58 @@ protected IActionResult BuildBadRequestErrorResponse(List<Error> errors)
                 CorrelationId = GetCurrentCorrelationId()
             });
         }
+        public class InternalServerErrorObjectResult : ObjectResult
+        {
+            public InternalServerErrorObjectResult(object value) : base(value)
+            {
+                StatusCode = StatusCodes.Status500InternalServerError;
+            }
+        }
+
+        protected IActionResult BuildInternalServerErrorResponse()
+        {           
+            return new InternalServerErrorObjectResult(new InternalServerError
+            {
+               CorrelationId = GetCurrentCorrelationId(),
+               Detail = InternalServerError,
+            });
+        }
+
+        public class NotModifiedObjectResult : ObjectResult
+        {
+            public NotModifiedObjectResult(object value) : base(value)
+            {
+                StatusCode = StatusCodes.Status304NotModified;
+            }
+        }
```

Comment:
Is there a reason for this class not to be extracted onto its own file?

Reply:
Previous thought about moving code in new class. However refactor code using same class.